File: Teacher-App/app/src/main/java/com/example/seeds/utils/LanguageUtils.kt
package com.example.seeds.utils

private val LANGUAGE_LABELS = mapOf(
    "en" to "English",
    "kn" to "Kannada",
    "hi" to "Hindi",
    "bn" to "Bengali",
    "ta" to "Tamil",
    "mr" to "Marathi",
    "or" to "Odia",
)

fun getLanguageLabel(iso: String): String =
    LANGUAGE_LABELS[iso.lowercase()] ?: iso.replaceFirstChar { it.uppercaseChar() }

File: Teacher-Expo-App/src/shared/utils/languageUtils.ts
const LANGUAGE_LABELS: Record<string, string> = {
  en: 'English',
  kn: 'Kannada',
  hi: 'Hindi',
  bn: 'Bengali',
  ta: 'Tamil',
  mr: 'Marathi',
  or: 'Odia',
};

export function getLanguageLabel(iso: string): string {
  if (!iso) return '';
  const lower = iso.toLowerCase();
  return LANGUAGE_LABELS[lower] ?? lower.charAt(0).toUpperCase() + lower.slice(1);
}
